### ISSUE TITLE
[Container cert] Require --docker-config option if --submit is specified; display which file is not found.

### ISSUE THREAD

**Issue by jfrancin** [redhat-openshift-ecosystem/openshift-preflight#685]
## Is your feature request related to a problem? Please describe.
At present in Preflight 1.2.1, if a user runs a container cert with the `--submit` option but leaves out the `--docker-config` option for a container whose repo is readable without authentication, the scan runs, but then fails with an inscrutable error message:

```
./preflight check container quay.io/purestorage/openstack-cinder-containers:16.1 --submit --pyxis-api-token=<HIDDEN> --certification-project-id=<PID>
time="2022-05-31T11:37:11-04:00" level=info msg="certification library version 1.2.1 <commit: 16e00a287ab183eb06e08839d0b87c5c7ab7c548>"
[...] 
time="2022-05-31T11:39:45-04:00" level=info msg="preparing results that will be submitted to Red Hat"
Error: open : no such file or directory
time="2022-05-31T11:39:47-04:00" level=fatal msg="open : no such file or directory"
```
This error is from the missing docker-config file.

So there are two problems here: 
a) The `--submit` option requires a `--docker-config` file option, but it's not enforced in command parsing.

b) File open error gives no clue about which file is missing.

## Describe the solution you'd like.

a) Enforce the requirement: if `--submit` is given but `--docker-config` is not, fail the command line parse. If you really want to be pedantic, check for the existence of the docker-config file during the parse rather than waiting until it tries to do the submit.

b) Beef up file open error message to indicate which file is missing.

**Comment by acornett21** [comment]
> a) The `--submit` option requires a `--docker-config` file option, but it's not enforced in command parsing.

This is used to be true that both were required, but now the user can give `PFLT_DOCKERCONFIG` which is why the `--docker-config` is not required with the `--submit` flag. Maybe we just need to add back that check but include an `|| os.Getenv("PFLT_DOCKERCONFIG").

**Comment by jfrancin** [comment]
That would work very well. One way or the other, the option should be satisfied.

**Comment by acornett21** [comment]
@bcrochet @komish  Two options:

- Add back in the logic to check for the `--docker-config` or `PFLT_DOCKERCONFIG` presence upfront and error out early
- Since `docker_config_json` is `nullable` in pyxis I'm thinking maybe we only add the dockerconfig to the project struct if `viper.GetString("dockerConfig")` is `""`. If we got to this point in the code, we were able to pull the image, which means it's a public repo.

I'm leaning towards the 2nd option, however we want to make sure we encourage partners with docker.io repos gives us a config because those requests from Clair are rate limited. Wondering what you all think?
